cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -841,39 +837,49 @@ void LocalVersionedEngine::delete_trees_responsibly(
     auto data_keys_not_to_be_deleted = get_data_keys_set(store(), *not_to_delete, read_opts);
     not_to_delete.clear();
     log::version().debug("Forbidden: {} total of data keys", data_keys_not_to_be_deleted.size());
-
     RemoveOpts remove_opts;
     remove_opts.ignores_missing_key_ = true;
+    
+    std::vector<entity::VariantKey> vks_column_stats;
+    std::transform(keys_to_delete->begin(),
+                    keys_to_delete->end(),
+                    std::back_inserter(vks_column_stats),
+                    [](const IndexTypeKey& index_key) {
+        return index_key_to_column_stats_key(index_key);
+    });
+    log::version().debug("Number of Column Stats keys to be deleted: {}", vks_column_stats.size());
 
-    std::vector<entity::VariantKey> vks;
-    // Delete any associated column stats keys first
-    if (!dry_run) {
-        std::transform(keys_to_delete->begin(),
-                       keys_to_delete->end(),
-                       std::back_inserter(vks),
-                       [](const IndexTypeKey& index_key) {
-            return index_key_to_column_stats_key(index_key);
-        });
-        store()->remove_keys(vks, remove_opts).get();
-    }
-
-    vks.clear();
-    if (!dry_run) {
-        std::copy(keys_to_delete->begin(), keys_to_delete->end(), std::back_inserter(vks));
-        store()->remove_keys(vks, remove_opts).get();
-    }
+    std::vector<entity::VariantKey> vks_to_delete;
+    std::copy(std::make_move_iterator(keys_to_delete->begin()),
+                std::make_move_iterator(keys_to_delete->end()),
+                std::back_inserter(vks_to_delete));
+    log::version().debug("Number of Index keys to be deleted: {}", vks_to_delete.size());
 
-    vks.clear();
+    std::vector<entity::VariantKey> vks_data_to_delete;
     std::copy_if(std::make_move_iterator(data_keys_to_be_deleted.begin()),
-                 std::make_move_iterator(data_keys_to_be_deleted.end()),
-                 std::back_inserter(vks),
-                 [&](const auto& k) {return !data_keys_not_to_be_deleted.count(k);});
-    log::version().debug("Index keys deleted. Proceeding with {} number of data keys", vks.size());
+                std::make_move_iterator(data_keys_to_be_deleted.end()),
+                std::back_inserter(vks_data_to_delete),
+                [&](const auto& k) {return !data_keys_not_to_be_deleted.count(k);});
+    log::version().debug("Number of Data keys to be deleted: {}", vks_data_to_delete.size());
+
+    folly::Future<folly::Unit> write_version_fut;
     if (!dry_run) {
-        store()->remove_keys(vks, remove_opts).get();
+        // Delete any associated column stats keys first
+        write_version_fut = store()->remove_keys(std::move(vks_column_stats), remove_opts)
+        .thenValue([this, vks_to_delete = std::move(vks_to_delete), remove_opts](auto&& ) mutable {
+            return store()->remove_keys(std::move(vks_to_delete), remove_opts);
```

Comment:
The logging has become slightly less useful. It used to log after the index keys were deleted, which gave some indication of progress. If you move the "Number of * keys to be deleted" logs into the `thenValue`s for index and data keys then it will show progress in the logs again.

Reply:
What's the point of having the dry_run option then?

Reply:
As discussed, separate logs have been added to track the progress of key removals.